### PR TITLE
pointerlock controls example : fix key listeners

### PR DESCRIPTION
**Pull request by felixmariotto**
Issue : the pointerlock example, FirstPersonControls and FlyControls checked the `keyCode` property of the `keydown` and `keyup` events, which means it was bound to **characters**. Some keyboard layouts, like the French one displayed bellow, don't map the WASD keys at the same place, whereas we actually play with the same keys as everybody, only these keys represent different characters.

Screenshot_1

The present PR solves this by using the [event.code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) propery, which is layout-independant. It will also solve the problem for Russian or Chinese layouts for instance.

### CODE REVIEW

**Review by Mugen87 — DISMISSED** [review]

**Comment by Mugen87** [comment]
It's probably best to only change `misc_controls_pointerlock` in this PR and clean up the code base when it was merged. 

Especially after a PR was approved, it's problematic to add more commits (since it invalidate existing the reviews).

**Comment by felixmariotto** [comment]
> It's probably best to only change misc_controls_pointerlock in this PR and clean up the code base when it was merged.

@Mugen87  Yes sorry, initially I just thought of fixing this example, but I found a lot more with problems... I'm done checking the examples now, I will do as you said 👍

File: examples/jsm/controls/FirstPersonControls.js [hunk 1]
```diff
@@ -155,45 +155,45 @@ var FirstPersonControls = function ( object, domElement ) {
 
 		//event.preventDefault();
 
-		switch ( event.keyCode ) {
+		switch ( event.code ) {
```
Comment (Mugen87) [inline examples/jsm/controls/FirstPersonControls.js:158]: You have to update the `examples/js` versions of the controls, too. Remember that modules are generated from the `examples/js` files via `node utils/modularize.js`.

**Comment by felixmariotto** [comment]
@Mugen87 I reverted to the state it was at when you reviewed it, and I will make a new PR for the rest of the changes, including the changes in `examples/js`.
Shall I squash the commits, or is it better to keep them separate ?

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
> Shall I squash the commits, or is it better to keep them separate ?

In this PR it seems you have force-pused which is totally fine.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**